src/abstractdataframe/reshape.jl

```diff
@@ -328,11 +328,14 @@ StackedVector(d::AbstractVector...)
 StackedVector(Any[[1,2], [9,10], [11,12]])  # [1,2,9,10,11,12]
 ```
 """
-struct StackedVector <: AbstractVector{Any}
+struct StackedVector{T} <: AbstractVector{T}
     components::Vector{Any}
 end
 
-function Base.getindex(v::StackedVector,i::Int)
+StackedVector(d::AbstractVector) =
+    StackedVector{promote_type(map(eltype, d)...)}(d)
+
+function Base.getindex(v::StackedVector, i::Int)
```

Comment:
Doesn't this need a `convert(T, ...)` for the returned value?

Reply:
Good point. This is consistent with `stack` without view and how `unstack` later works. I will add this change.
 
(the alternative would be to use `typejoin` not `promote_type` in line 336. Then we do not need to convert, but it would not be consistent with `stack` that materializes.)